### PR TITLE
vmware_host: Small fixes and docs updates

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This PR includes:
- A fix to no longer require a datacenter folder for adding a host
- Documentation improvements
- Ensure imports are specific

This fixes #26326 (actually it includes @Akasurde 's changes too, see log below)

Fix adds following:
* Update logic in vmware_host
* Update example documentation
* Added test case for vmware_host

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE

 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME

vmware_host

##### ANSIBLE VERSION
v2.4

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_host.py [hunk 1]
```diff
@@ -131,7 +140,10 @@ def process_state(self):
 
     def find_host_by_cluster_datacenter(self):
         self.dc = find_datacenter_by_name(self.content, self.datacenter_name)
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host.py:142]: if self.dc is None:

File: lib/ansible/modules/cloud/vmware/vmware_host.py [hunk 2]
```diff
@@ -22,67 +22,66 @@
...
+    - ESXi username.
+    required: yes
+  esxi_password:
+    description:
+    - ESXi password.
+    required: yes
+  state:
+    description:
+    - Add or remove the host.
+  choices: [absent, present]
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host.py]: Indent is required here and below line.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_host.py (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/modules/cloud/vmware/vmware_host.py]: Yeah, I know, is a man allowed to sleep around here ? :-D

File: lib/ansible/modules/cloud/vmware/vmware_host.py (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/cloud/vmware/vmware_host.py:142]: Nope, that's the whole point of my PR, self.dc can be None !

**Review by Akasurde — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
shipit

**Pull request merged**